core/native/src/files/PathsNative.kt

```diff
@@ -16,17 +16,19 @@ import platform.posix.*
  */
 
 public actual class Path internal constructor(
-    internal val path: String,
+    rawPath: String,
     @Suppress("UNUSED_PARAMETER") any: Any?
 ) {
+    internal val path = removeTrailingSeparators(rawPath)
+
     public actual val parent: Path?
         get() {
             when {
-                path.isBlank() -> return null
+                path.isEmpty() -> return null
```

Comment:
why aren't checks `!path.contains(` used like in JS?

Reply:
On native, that check is performed by corresponding `dirnameImpl` implementations. 
For example, https://github.com/Kotlin/kotlinx-io/blob/fe9eefc72eb2f500ab7f1f0cd5f7d1baa351d815/core/linux/src/files/FileSystemLinux.kt#L17
On all native platforms except MinGW, it checks `/` presence, but on MinGW, it also checks for `\` presence. For that reason, the check is inside `dirnameImpl` and not inside `parent`.